### PR TITLE
Refactor svg and svg all the things

### PR DESCRIPTION
**Pull request by paulfalgout**
Somewhat of a large PR.

- I optimized the svg files and svg'd the marionette logo and the mojotech logo.
- I inlined the back-to-top svg in the css
- I removed the play.png from the videos and utilize a youtube similar play button svg
- The rest of the svgs are now in `src/svg-icons/` and during build they get added to `src/images/svg-sprite.svg` which in turn is inlined at the top of the html files that use the icons.

Then in the html to use an svg you would:
```
<svg>
  <use xlink:href="#filename">
</svg>
```
where `#filename` === `src/svg-icons/filename.svg`

You can then use css to set the color or size of the svg or even colors of the elements of the svg.  All of the current svgs are using `fill="currentColor"` so they will inherit the font color of it's parent element, but this is a convenience that doesn't have to be maintained in the future.

This svg setup is somewhat outlined here: http://css-tricks.com/svg-sprites-use-better-icon-fonts/

There's a slight learning curve to adding a new svg icon, but less so than a font icon.  Also since the sprite has to be inlined in the html it's not cached, however these end up being very very small after the html is gzipped, but the upside there's no additional file transfers.

Resolves https://github.com/marionettejs/marionettejs.com/issues/97

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,6 +45,7 @@
     "grunt-notify": "0.4.1",
     "grunt-postcss": "0.2.0",
     "grunt-sass": "0.17.0",
+    "grunt-svgstore": "^0.5.0",
```
Comment (samccone) [inline package.json]: mind :lock:

File: src/docs/template.html [hunk 2]
```diff
@@ -4,21 +4,30 @@
...
           <a href="https://twitter.com/marionettejs" title="Twitter">
-              <img src="../../images/twitter.svg">
+              <svg width="18px" height="18px">
+                <use xlink:href="#twitter-square">
+              </svg>
           </a>
           <a href="https://gitter.im/marionettejs/backbone.marionette" title="Gitter">
-              <img src="../../images/gitter.svg">
+              <svg width="18px" height="18px">
+                <use xlink:href="#gitter-square">
```
Comment (samccone) [inline src/docs/template.html]: this is s coooool

File: src/stylesheets/_back_to_top_button.scss [hunk 3]
```diff
@@ -18,7 +18,7 @@
   text-indent: 100%;
   white-space: nowrap;
 
-  background: rgba($med-gray, .8) url(../images/top-arrow.svg) no-repeat center 50%;
+  background: rgba($med-gray, .8) url('data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16"><path fill="#fff" d="M8 2.8l8 7.9-2.4 2.4-5.5-5.5-5.6 5.6L0 10.7z"/></svg>') no-repeat center 50%;
```
Comment (samccone) [inline src/stylesheets/_back_to_top_button.scss]: :)

**Comment by paulfalgout** [comment]
script lock'd

**Comment by samccone** [comment]
:clap:

**Pull request merged**